### DISCUSSION TITLE
Getting recording state and handling of SIGUSR1 and SIGUSR2

### DISCUSSION THREAD

**Discussion by wschell** [jamulussoftware/jamulus#1953] [Ideas]
By now SIGUSR1 starts a new recording - but only if recording is already active. SIGUSR2 toggles recording state .

But when scripting it is not easy to determine the actual recording state. There is no entry in the log file (--log parameter)  and no info in the html status page. One can check if files in the recording directory are in use (lsof) or look into syslog where the message from all servers started with systemd are logged. 

It would be great to have the recording state also in the file given by the --log parameter and eventually in the html status page.

Additionally -  it seems that by now SIGUSR1 and SIGUSR2 perform actions for special cases. 
For me it would be more intuitive, if one signal only starts/restarts recording and the other signal only stops recording. So one knows exactly which recording state is actual. It will need an additional start parameter to not break the actual behaviour.

**Comment by npostavs** [comment]
See also #1789 for adding a more general mechanism than signals for talking to the server.

**Reply by wschell** [reply]
That's of course a great idea. But this will be a big change to source code and sure will take some time and a lot of testing.
 
Optional changing the meaning of SIGUSR1 and SIGUSR2 would only need few lines of code and could be done very fast. But it seems, that no one else has problems with the actual implementation of SIGUSR1 and SIGUSR2 so ...
